client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -0,0 +1,157 @@
+) => {
+	return translate( '{{price/}} /mailbox {{intervalLength/}}', {
+		components: {
+			price: (
+				<span className={ formattedPriceClassName }>
+					{ formatCurrency( cost ?? 0, currencyCode ) }
+				</span>
+			),
+			intervalLength: <span className={ intervalLengthClassName }>/{ intervalLength }</span>,
+		},
+		comment:
+			'{{price/}} is the formatted price, e.g. $20' +
+			'{{intervalLength/}} is already translated and it is either annually or monthly',
+	} );
+};
+
+export const addToCartAndCheckout = (
+	shoppingCartManager: ShoppingCartManagerActions,
+	cartItem: RequestCartProduct | IncompleteRequestCartProduct,
+	productList: Record< string, { product_id: number } >,
+	setAddingToCart: ( addingToCart: boolean ) => void,
+	selectedSite: string
+): void => {
+	setAddingToCart( true );
+
+	shoppingCartManager
+		.addProductsToCart( [ fillInSingleCartItemAttributes( cartItem, productList ) ] )
+		.then( ( response: ResponseCart ) => {
+			setAddingToCart( false );
+			const errors = response?.messages?.errors;
+			if ( errors && errors.length ) {
+				// Stay on the page to show the relevant error(s)
+				return;
+			}
+
+			page( `/checkout/${ selectedSite }` );
+		} )
+		.then( () => {
+			setAddingToCart( false );
+		} );
+};
+
+export const recordTracksEventAddToCartClick = (
+	comparisonContext: string,
+	validatedMailboxUuids: string[],
+	mailboxesAreValid: boolean,
+	provider: string,
+	source: string,
+	userCanAddEmail: boolean,
+	userCannotAddEmailReason: any
+) => {
+	recordTracksEvent( 'calypso_email_providers_add_click', {
+		context: comparisonContext,
+		mailbox_count: validatedMailboxUuids.length,
+		mailboxes_valid: mailboxesAreValid ? 1 : 0,
+		provider: provider,
+		source,
+		user_can_add_email: userCanAddEmail,
+		user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
+	} );
+};
+
+type PriceWithIntervalProps = {
+	className: string;
+	cost: number;
+	currencyCode: string;
+	hasDiscount: boolean;
+	sale?: number | null;
+	intervalLength: IntervalLength;
+};
+
+export const PriceWithInterval: FunctionComponent< PriceWithIntervalProps > = ( props ) => {
+	const { className, cost, currencyCode, hasDiscount, sale, intervalLength } = props;
+
+	const costClassName = `${ className }__${ hasDiscount ? 'discounted-price' : 'keep-main-price' }`;
+	const intervalLengthClassName = `${ className }__interval`;
+	const mailboxClassName = `${ className }__mailbox`;
+	const saleClassName = `${ className }__sale`;
+
+	const showSale = sale && sale !== 0;
+
+	const lineThrough = {
+		textDecoration: hasDiscount || showSale ? 'line-through' : 'solid ',
+	};
+
+	const priceSpan = (
+		<span style={ lineThrough } className={ costClassName }>
+			{ formatCurrency( cost ?? 0, currencyCode ) }
+		</span>
+	);
+	const saleSpan = (
+		<span className={ saleClassName }>{ formatCurrency( sale ?? 0, currencyCode ) }</span>
+	);
+	const intervalLengthSpan =
+		intervalLength === 'annually' ? (
```

Comment:
We should use the constant here:

```suggestion
		intervalLength === IntervalLength.ANNUALLY ? (
```